### ISSUE TITLE
Legend for visualizeRegion/visualizeGene/visualizeProbes

### ISSUE THREAD

**Issue by alanlamsiu** [zwdzwd/sesame#79]
Is there a way to add the legend (color scale) when using visualizeRegion/visualizeGene/visualizeProbes? Thanks!

**Comment by zwdzwd** [comment]
Yes, you can add it with WLegendV or WLegendH to the resulting grid objects. something like
```
visualizeProbes(...) + WLegendV('betas', TopRightOf())
```

**Comment by alanlamsiu** [comment]
Thanks for pointing to the WLegendV function. I tried it with visualizeGene() but it showed the "Error in objs[[1]] : subscript out of bounds" message. Do you think it has something to do with the betas object, which was generated by openSesame()?

**Comment by zwdzwd** [comment]
Can you try the following example?
```
betas <- sesameDataGet('HM450.10.TCGA.PAAD.normal')
visualizeGene('DNMT1', betas, platform='HM450') + WLegendV("betas", TopRightOf("betas"))
```

**Comment by alanlamsiu** [comment]
Yes, this example works fine. I noticed that there were some probes with NAs and I thought thoese might be an issue. So I removed probes with NA, however, the subscript out of bounds error still occurred. If not adding WLegendV(), the figure was generated without any issue.

**Comment by zwdzwd** [comment]
One possibility is that there is no beta value heatmap after removing the NAs. If so, yeah, the legend might be struggling to find the original color scale.

**Comment by alanlamsiu** [comment]
I would also suspect so. Below is what the heatmap looks like after removing the NAs. Yet, the "subscript out of bound" error persisted.
![image](https://user-images.githubusercontent.com/34832128/186325273-ee71540b-41dc-4e54-988f-2fb384963ae7.png)

**Comment by zwdzwd** [comment]
OK. If you can share with me an example to reproduce this, I can troubleshoot more.

**Comment by alanlamsiu** [comment]
I extracted the beta values of 16 probes associated with the FMR1 gene as shown in this table. I tried to use this matrix as input for visualizeGene() and WLegendV() but the error message showed up on my end. Appreciate if you could look into the issue.
[test_betas.txt](https://github.com/zwdzwd/sesame/files/9416994/test_betas.txt)

**Comment by zwdzwd** [comment]
can you try
```
betas = read_tsv("~/Downloads/test_betas.txt")
b2 = as.matrix(betas[,2:7])
rownames(b2) = betas[[1]]
visualizeGene("FMR1", b2, platform="EPIC") + WLegendV("betas", TopRightOf("betas"))
```
